Samples/Samples/View/CalendarEventPage.xaml.cs

```diff
@@ -29,28 +38,41 @@ async void OnDeleteEventButtonClicked(object sender, EventArgs e)
             if (!(calendarEvent is CalendarEvent))
                 return;
 
-            var answer = await DisplayAlert("Warning!", $"Are you sure you want to delete {calendarEvent.Title}? (this cannot be undone)", "Yes", "Cancel");
+            var answer = await DisplayAlert(actionTitleWarning, $"Are you sure you want to delete {calendarEvent.Title}? (this cannot be undone)", actionResponseYes, actionResponseCancel);
             if (answer)
             {
                 if (calendarEvent.RecurrancePattern != null)
                 {
-                    if (await DisplayAlert("Warning!", $"Do you want to delete all instances of this event?", "Yes All", "Just this one"))
-                    {
-                        if (await Calendars.DeleteCalendarEventById(eventId, CalendarId.Text))
-                        {
-                            await DisplayAlert("Info", "Deleted event id: " + eventId, "Ok");
-                            await Navigation.PopAsync();
-                        }
-                    }
-                    else if (await Calendars.DeleteCalendarEventInstanceByDate(eventId, CalendarId.Text, calendarEvent.StartDate))
+                    var action = await DisplayActionSheet("Do you want to delete all instances of this event?", actionResponseCancel, null, actionResponseDeleteAll, actionResponseDeleteOne, actionResponseDeleteForward);
+
+                    switch (action)
                     {
-                        await DisplayAlert("Info", "Deleted event id: " + eventId, "Ok");
-                        await Navigation.PopAsync();
+                        case actionResponseDeleteAll:
+                            if (await Calendars.DeleteCalendarEventById(eventId, CalendarId.Text))
+                            {
```

Comment:
These if blocks all look similar. Suggest have the following declaration prior to the switch
var deletionConfirmedMessage=string.Empty;
in each of the if blocks just set the deletionConfirmedMessage
After the switch
if(!string.isNullOrwhitespace(deletionConfirmedMessage)){
 await Dialog...
 await Navigation.PopAsync
}
else{
await Dialog ("Unable to delete.....)
}

Reply:
Altered accordingly